### PR TITLE
Ubuntu 18.04 Installation and Testing Scripts

### PR DESCRIPTION
**Pull request by anshuljainansja**

### CODE REVIEW

File: ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh [hunk 1]
```diff
@@ -1,14 +1,12 @@
 #!/bin/bash
 set -ex
 
-# Change versions (Ubuntu, GPU driver) here if needed
-# See https://github.com/Azure/azhpc-extensions/blob/master/NvidiaGPU/resources.json
-DRIVER_URL=https://go.microsoft.com/fwlink/?linkid=874271
-CUDA_REPO_PKG=cuda-repo-ubuntu_amd64.deb
-PUBKEY_URL=http://download.microsoft.com/download/F/F/A/FFAC979D-AD9C-4684-A6CE-C92BB9372A3B/7fa2af80.pub
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin
+mv cuda-ubuntu1804.pin /etc/apt/preferences.d/cuda-repository-pin-600
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh:5]: We may want to unify this with how extensions are installed.

File: ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-sudo dpkg -i $CUDA_REPO_PKG
-sudo apt-key adv --fetch-keys $PUBKEY_URL 
-sudo apt-get install --no-install-recommends -y cuda-drivers
-sudo apt-get install -y cuda
+wget http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.33.01_1.0-1_amd64.deb
+
+dpkg -i cuda-repo-ubuntu1804-10-2-local-10.2.89-440.33.01_1.0-1_amd64.deb
+apt-key add /var/cuda-repo-10-2-local-10.2.89-440.33.01/7fa2af80.pub
+apt-get update
+apt-get -y install cuda
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh:12]: how about cuda-drivers?

Comment (vermagit) [reply ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh:12]: Yes, cuda-drivers too. Thanks.

File: ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh (hunk 1 above)
Comment (vermagit) [reply ubuntu/ubuntu-18.04-hpc/install_nvidiagpudriver.sh:5]: Yes, as I mentioned in the seed PR, the end objective is to unify the extension with the scripts here. I can talk about how that can be achieved. But basically that encompasses not having multiple scripts in different places doing the same thing.
As a first pass and PoC on the Ubuntu image, I am fine with us not tackling that issue yet.

However, why should we remove the variables? Also, the "pin" method above is different than manual steps we publish and extension. Any advantage with this?

File: tests/run-tests.sh [hunk 3]
```diff
@@ -99,46 +155,78 @@ check_exists "/opt/amd/fftw/"
...
-module load mpi/impi-2019
-mpiexec -np 2 -ppn 2 -env FI_PROVIDER=mlx -env UCX_TLS=rc -env I_MPI_SHM=0 ${IMPI2019_PATH}/linux/mpi/intel64/bin/IMB-MPI1 pingpong
-check_exit_code "Intel MPI 2019" "Failed to run Intel MPI 2019"
-module unload mpi/impi-2019
+if [ $CHECK_IMPI_2019 -eq 1 ]
+then
+    check_exists "${MODULE_FILES_ROOT}/mpi/impi-2019"
+
+    module load mpi/impi-2019
+    mpiexec -np 2 -ppn 2 -env FI_PROVIDER=verbs -env UCX_TLS=rc -env I_MPI_SHM=0 ${IMPI2019_PATH}/linux/mpi/intel64/bin/IMB-MPI1 pingpong
```
Comment (jithinjosepkl) [inline tests/run-tests.sh:178]: -env UCX_TLS=rc is not needed here.

**Pull request merged**